pubsub/rabbitmq/rabbitmq.go

```diff
@@ -563,17 +561,19 @@ func (r *rabbitMQ) reset() (err error) {
 }
 
 func (r *rabbitMQ) isStopped() bool {
-	return r.ctx.Err() != nil
+	select {
+	case <-r.closeCh:
```

Comment:
If the channel is just used for this, consider using an atomic.Bool to use a semaphore. Much simpler than relying on properties like "reading from a closed channel always returns a nil value immediately" :)